packages/material-ui/src/StepButton/StepButton.test.js

```diff
@@ -1,122 +1,131 @@
 import React from 'react';
-import { assert } from 'chai';
+import { expect } from 'chai';
 import { spy } from 'sinon';
-import { createShallow, createMount, getClasses } from '@material-ui/core/test-utils';
+import { createMount, getClasses } from '@material-ui/core/test-utils';
 import describeConformance from '../test-utils/describeConformance';
+import { createClientRender } from 'test/utils/createClientRender';
 import StepButton from './StepButton';
+import Step from '../Step';
 import StepLabel from '../StepLabel';
 import ButtonBase from '../ButtonBase';
+import { fireEvent } from '@testing-library/dom';
 
 describe('<StepButton />', () => {
   let classes;
-  let shallow;
-  let mount;
-  const defaultProps = { orientation: 'horizontal' };
+  const render = createClientRender();
 
   before(() => {
     classes = getClasses(<StepButton />);
-    shallow = createShallow({ dive: true });
-    mount = createMount({ strict: true });
   });
 
-  after(() => {
-    mount.cleanUp();
-  });
+  describe('internals', () => {
+    let mount;
 
-  describeConformance(<StepButton {...defaultProps} />, () => ({
-    classes,
-    inheritComponent: ButtonBase,
-    mount,
-    refInstanceof: window.HTMLButtonElement,
-    skip: ['componentProp'],
-  }));
-
-  it('passes active, completed, disabled to StepLabel', () => {
-    const wrapper = shallow(
-      <StepButton active completed disabled {...defaultProps}>
-        Step One
-      </StepButton>,
-    );
-    const stepLabel = wrapper.find(StepLabel);
-    assert.strictEqual(stepLabel.props().active, true);
-    assert.strictEqual(stepLabel.props().completed, true);
-    assert.strictEqual(stepLabel.props().disabled, true);
-    assert.strictEqual(stepLabel.props().children, 'Step One');
-  });
+    before(() => {
+      mount = createMount({ strict: true });
+    });
 
-  it('should pass props to a provided StepLabel', () => {
-    const wrapper = shallow(
-      <StepButton active completed disabled label="Step One" {...defaultProps}>
-        <StepLabel>Step One</StepLabel>
-      </StepButton>,
-    );
-    const stepLabel = wrapper.find(StepLabel);
-    assert.strictEqual(stepLabel.props().active, true);
-    assert.strictEqual(stepLabel.props().completed, true);
-    assert.strictEqual(stepLabel.props().disabled, true);
-  });
+    after(() => {
+      mount.cleanUp();
+    });
 
-  it('should pass disabled prop to the ButtonBase', () => {
-    const wrapper = shallow(
-      <StepButton disabled {...defaultProps}>
-        Step One
-      </StepButton>,
-    );
-    const stepLabel = wrapper.find(ButtonBase);
-    assert.strictEqual(stepLabel.props().disabled, true);
-  });
+    describeConformance(<StepButton />, () => ({
+      classes,
+      inheritComponent: ButtonBase,
+      mount,
+      refInstanceof: window.HTMLButtonElement,
+      skip: ['componentProp'],
+    }));
 
-  describe('event handlers', () => {
-    describe('handleMouseEnter/Leave', () => {
```

Comment:
This handles the same as the "should bubble callbacks used internally" test whose description is misleading. It never tested bubbling as far as I can tell.